src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java

```diff
@@ -42,18 +44,29 @@
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.nio.charset.Charset;
+import org.kohsuke.stapler.DataBoundSetter;
 
 /**
  * Runs a Powershell script
  */
 public final class PowershellScript extends FileMonitoringTask {
     private final String script;
+    private String powershellBinary = "powershell";
     private boolean capturingOutput;
     
     @DataBoundConstructor public PowershellScript(String script) {
         this.script = script;
     }
-    
+
+    public String getPowershellBinary() {
+        return powershellBinary;
+    }
+
+    @DataBoundSetter
+    public void setPowershellBinary(String powershellBinary) {
+        this.powershellBinary = powershellBinary;
```

Comment:
The UI only allows two options however from API it could by anything.

Do you see an issue with that?

Reply:
Looking into it some more, since https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/122 is the only way users would be exposed to the change, and they wouldn't specify the actual binary, just the new `pwsh` step, I don't think validation is necessary. 

It seems a bit confusing that we even have Jelly files in this plugin, since IIUC they are not currently used anywhere, because the steps in workflow-durable-step define their own config and help files?

Reply:
The jelly files here should appear in FreeStyle but as I wrote in the PR description. This does not currently work for master branch nor this PR.

Should I try and do a git bisect to determine when PowerShell option disappeared from Jelly view?

Reply:
I thought that feature was provided by this plugin: https://github.com/jenkinsci/powershell-plugin?

Reply:
it was, welp that's good good to know. Let me send a downstream PR their way to test.

Reply:
oh they are not using the durable task, okay even more confusing. So PowershellScript is only here to please workflow-durable-task-step?

Reply:
Yeah, at least currently, the only consumer of the APIs here is workflow-durable-task-step, that's why it seems confusing that we even have Jelly files here at all, because they are not used by any existing plugins.

Reply:
in any case, I would still appreciate a release to finally add pwsh as a pipeline step.

sym linking is not a good workaround, especially since you cannot directly use a PowerShell core docker image.

Reply:
Yeah, of course, all I meant was that if users were specifying `powershellBinary` directly, it might be good to add some validation like this:

```
if (!(powershellBinary == null || powershellBinary.equalsIgnoreCase("powershell") || powershellBinary.equalsIgnoreCase("pwsh"))) {
    throw new IllegalArgumentException("...")
}
```

But since they aren't specifying the binary, they are just picking whether to use the `powershell` or `pwsh` step, there's no reason to add validation.

Reply:
Sure, do you mind if I switch the `powershell` and `pwsh` to public static final string or do you prefer enum?

Reply:
I'm not sure what you mean, I think the code in the PR is good as-is, no changes necessary. I'm happy to merge it once the CI build passes.

Reply:
Okay 👍

Reply:
It looks like the CI build is unavailable?  Getting a 404 page.

Reply:
Yeah, I think ci.jenkins.io is down right now (unfortunately a common occurrence these days). Looks like the infra team is investigating based on #jenkins-infra on IRC.